--- src/org/traccar/protocol/AquilaProtocolDecoder.java
+++ src/org/traccar/protocol/AquilaProtocolDecoder.java
@@ -124,9 +124,12 @@
                 (Integer.parseInt(parser.group(index++)) << 1) +
                 (Integer.parseInt(parser.group(index++)));
 
+        try {
         if (course > 0 && course <= 8) {
             position.setCourse((course - 1) * 45);
         }
+		} catch (Throwable e) {
+		}
 
         return position;
     }
