--- src/main/java/com/fasterxml/jackson/databind/util/StdDateFormat.java
+++ src/main/java/com/fasterxml/jackson/databind/util/StdDateFormat.java
@@ -523,11 +523,8 @@
                 df = _formatISO8601_z;
                 formatStr = DATE_FORMAT_STR_ISO8601_Z;
                 if (df == null) {
-                    // 10-Jun-2017, tatu: As per [databind#1651], when using this format,
-                    //    must use UTC, not whatever is configured as default timezone
-                    //    (because we know `Z` identifier is used)
-                    df = _formatISO8601_z = _cloneFormat(DATE_FORMAT_ISO8601_Z, formatStr,
-                            DEFAULT_TIMEZONE, _locale, _lenient);
+                    df = _formatISO8601_z = _cloneFormat(DATE_FORMAT_ISO8601_Z, formatStr, _timezone, _locale,
+							_lenient);
                 }
             }
         }
