--- src/main/java/org/openpnp/util/OpenCvUtils.java
+++ src/main/java/org/openpnp/util/OpenCvUtils.java
@@ -239,6 +239,9 @@
     }
 
     public static void saveDebugImage(Class implementationClass, String function, String identifier, Mat mat) {
+        try {
         saveDebugImage(implementationClass, function, identifier, OpenCvUtils.toBufferedImage(mat));
+		} catch (Throwable e) {
+		}
     }
 }
