--- src/main/java/org/stagemonitor/core/metrics/metrics2/Metric2RegistryModule.java
+++ src/main/java/org/stagemonitor/core/metrics/metrics2/Metric2RegistryModule.java
@@ -164,9 +164,9 @@
 		writeDoubleUnlessNaN(jg, "min", convertDuration(snapshot.getMin()));
 		writeDoubleUnlessNaN(jg, "max", convertDuration(snapshot.getMax()));
 		writeDoubleUnlessNaN(jg, "mean", convertDuration(snapshot.getMean()));
+		writeDoubleUnlessNaN(jg, "median", convertDuration(snapshot.getMedian()));
 		writeDoubleUnlessNaN(jg, "std", convertDuration(snapshot.getStdDev()));
 		writeDoubleUnlessNaN(jg, "p25", convertDuration(snapshot.getValue(0.25)));
-		writeDoubleUnlessNaN(jg, "p50", convertDuration(snapshot.getMedian()));
 		writeDoubleUnlessNaN(jg, "p75", convertDuration(snapshot.get75thPercentile()));
 		writeDoubleUnlessNaN(jg, "p95", convertDuration(snapshot.get95thPercentile()));
 		writeDoubleUnlessNaN(jg, "p98", convertDuration(snapshot.get98thPercentile()));
