--- src/org/traccar/protocol/EelinkProtocolDecoder.java
+++ src/org/traccar/protocol/EelinkProtocolDecoder.java
@@ -160,7 +160,10 @@
 
             if (statusType == 0x01 || statusType == 0x02 || statusType == 0x03) {
                 buf.readUnsignedInt(); // device time
+                try {
                 decodeStatus(position, buf.readUnsignedShort());
+				} catch (Throwable e) {
+				}
             }
 
         }
