--- src/org/traccar/protocol/EelinkProtocolDecoder.java
+++ src/org/traccar/protocol/EelinkProtocolDecoder.java
@@ -160,7 +160,7 @@
 
             if (statusType == 0x01 || statusType == 0x02 || statusType == 0x03) {
                 buf.readUnsignedInt(); // device time
-                if (statusType == 0x02) {
+                if (type == MSG_ALARM) {
 					decodeStatus(position, buf.readUnsignedShort());
 				}
             }
