--- src/main/java/net/bytebuddy/dynamic/scaffold/InstrumentedType.java
+++ src/main/java/net/bytebuddy/dynamic/scaffold/InstrumentedType.java
@@ -880,7 +880,6 @@
                 } else if (methodDescription.isConstructor()) {
                     TypeDescription declaringType = getDeclaringType();
                     if (!methodDescription.getDeclaringType().equals(declaringType == null ? this : declaringType)) {
-                        throw new IllegalStateException("Constructor " + methodDescription + " defines an illegal receiver " + receiverType);
                     }
                 } else if (!equals(receiverType)) {
                     throw new IllegalStateException("Method " + methodDescription + " defines an illegal receiver " + receiverType);
