--- src/main/java/com/webfirmframework/wffweb/tag/html/AbstractHtml.java
+++ src/main/java/com/webfirmframework/wffweb/tag/html/AbstractHtml.java
@@ -1114,6 +1114,7 @@
         }
 
         attributesMap = new ConcurrentHashMap<>(attributes.length);
+		attributesMap=new ConcurrentHashMap<>();
 
         for (final AbstractAttribute attribute : attributes) {
             attributesMap.put(attribute.getAttributeName(), attribute);
