--- src/org/traccar/BaseProtocolDecoder.java
+++ src/org/traccar/BaseProtocolDecoder.java
@@ -106,8 +106,7 @@
     }
 
     public DeviceSession getDeviceSession(Channel channel, SocketAddress remoteAddress, String... uniqueIds) {
-        if (channel.getPipeline().get(HttpRequestDecoder.class) != null
-                || Context.getConfig().getBoolean("decoder.ignoreSessionCache")) {
+        if (Context.getConfig().getBoolean("decoder.ignoreSessionCache")) {
             long deviceId = findDeviceId(remoteAddress, uniqueIds);
             if (deviceId != 0) {
                 if (Context.getConnectionManager() != null) {
